### DISCUSSION TITLE
v.0.2.0

### DISCUSSION THREAD

**Discussion by jpsacha** [ij-plugins/ijp-imagej-launcher#9] [Announcements]
This is a feature release. One of the most important improvements is a support for better system integration with macOS. Now ImageJ/Fiji can be installed in the Application directory. See macOS installation info in the [ReadMe](https://github.com/ij-plugins/ijp-imagej-launcher)

All improvements:
* Use launcher location to infer `ij-dir`. #5
* Session log is saved to `~/.ijp_imagej_launcher.log` to facilitate troubleshooting when not running from command
  prompt. The log is reset for each session. #6
* Better inference of ImageJ directory on macOS - consider launcher being in subdirectory "Contents/MacOS" #7 

<hr /><em>This discussion was created from the release <a href='https://github.com/ij-plugins/ijp-imagej-launcher/releases/tag/v.0.2.0'>v.0.2.0</a>.</em>

**Comment by emma2117** [comment]
following your mac installation instructions in read.me gave an error message when trying to finally launch fiji at the end - "Fiji is damaged and you should move to trash"

**Reply by jpsacha** [reply]
I never seen such an error message. I would start the process from the beginning, making sure that no changes are made to Fiji.app folder except replacing `ImageJ-macosx` and adding `java` folder, as described in the [ReadMe](https://github.com/ij-plugins/ijp-imagej-launcher#installing-fiji-on-mac-os-x-arm64)

**Reply by Kim-KL** [reply]
I experienced the same problem now. I don't know what happen. My macOS version is 13.5.1

**Reply by jpsacha** [reply]
Can you confirm that the message is exactly "Fiji is damaged and you should move to trash" or is it worded differently?

**Reply by Kim-KL** [reply]
It was exactly same. "Fiji is damaged and you should move to trash"

**Reply by jpsacha** [reply]
Are you getting the message for the Fiji.app without modifications? That is if you download Fiji, uncompressed it and then click on "Fiji.app" folder are you getting the same message?

**Reply by eppendork** [reply]
Can confirm I was also obtaining the same error. However, I was able to fix this by running the No JRE Fiji.app and allowing unidentified developer permission prior to adding Java runtime and ImageJ launcher to the app. Hope this helps!

**Reply by ctrueden** [reply]
One way to fix the issue is to use some Terminal commands to remove the quarantine flag from the downloaded Fiji folder. See [this topic on forum.image.sc](https://forum.image.sc/t/warning-your-imagej-installation-cannot-be-updated-because-it-is-read-only/20333/10?u=ctrueden).

**Comment by VerschuurenM** [comment]
Following your instructions, the installation was successful. However, when trying to access the tensor flow menu (Edit > Options > Tensorflow..) it evoked an error.

**Reply by jpsacha** [reply]
@VerschuurenM Can you describe what the error is?

**Comment by MontseDroguett** [comment]
Hi! Any chance you can release an IJP-ImageJ-Launcher for MacOs x64 (intel)?

**Reply by ctrueden** [reply]
I don't know @jpsacha's plans for IJP-ImageJ-Launcher, but I would encourage you to try Jaunch, the upcoming new launcher for Fiji which I am currently developing. You can download a prerelease build for all platforms (including both macOS Intel and ARM64) from [here](https://downloads.imagej.net/~curtis/jaunch-fiji-launcher-20240213.zip). Unpack into an existing Fiji.app to put everything into the right places. For more details, see [this forum post](https://forum.image.sc/t/jaunch-a-new-java-launcher-test-fiji-with-java-21/92058).

**Reply by jpsacha** [reply]
I am waiting for updates to the project dependencies/compiling tools that could improve the launch process monitoring and error reporting. The key change was new multiprocessing support in scala-native. It was release recently. In turn that requires updates to os-lib (released recently too) and scopt (it is in process of being updated, and hoping for a new new release soon). 

Hopefully Jaunch will work for you. If not, let me know after the next `ijp-imagej-launcher` release. MacOS Intel is tricker for me to test.

**Reply by MontseDroguett** [reply]
I will try Jaunch, thank you for your responses! :)

**Comment by Stellanliu** [comment]
I followed your mac installation instructions in the troubleshooting and moved it into the application. When I tried to open it, the dialog tells that the application requires macOS 13.0 or later. But I'm on macOS 12.7.6 monterey. Is the system requirement mandatory?

**Comment by jpsacha** [comment]
There is no explicit requirement in ijp-imagej-launcher for a specific
macOS version. Not clear to me what could be causing the “ requires macOS
13.0” message.
